sky/packages/sky/lib/src/widgets/basic.dart

```diff
@@ -513,8 +513,60 @@ class BlockBody extends MultiChildRenderObjectWidget {
 }
 
 class Stack extends MultiChildRenderObjectWidget {
-  Stack(List<Widget> children, { Key key }) : super(key: key, children: children);
-  RenderStack createRenderObject() => new RenderStack();
+  Stack(List<Widget> children, {
+    Key key,
+    this.horizontalAlignment: 0.0,
+    this.verticalAlignment: 0.0
+  }) : super(key: key, children: children) {
+    assert(horizontalAlignment != null);
+    assert(verticalAlignment != null);
+  }
+
+  final double horizontalAlignment;
+  final double verticalAlignment;
+
+  RenderStack createRenderObject() {
+    return new RenderStack(
+      horizontalAlignment: horizontalAlignment,
+      verticalAlignment: verticalAlignment
+    );
+  }
+
+  void updateRenderObject(RenderStack renderObject, Stack oldWidget) {
+    renderObject.horizontalAlignment = horizontalAlignment;
+    renderObject.verticalAlignment = verticalAlignment;
+  }
+}
+
+class IndexedStack extends Stack {
+  IndexedStack(List<Widget> children, {
+    Key key,
+    double horizontalAlignment: 0.0,
+    double verticalAlignment: 0.0,
+    this.index: 0
+  }) : super(children,
+    key: key,
+    horizontalAlignment: horizontalAlignment,
+    verticalAlignment: verticalAlignment
+  ) {
+    assert(index >= 0);
+    assert(children != null ? index < children.length : true);
```

Comment:
so index can be out of range if children is null but -1 if children is empty?
Maybe index should be -1 or null if children is null or empty?
Note that as written, these asserts preclude an empty list.